smeargle.py

```diff
@@ -177,103 +184,109 @@ def sd_api_call():
+        print(f"Current Directory: {dirpath}")
+        print(f"Found Files: {files}")
+        for musicfile in files:
+            if musicfile.endswith(".mp3"):
+                check_and_generate(dirpath, musicfile, music_extension=".mp3")
+            elif musicfile.endswith(".opus"):
+                check_and_generate(dirpath, musicfile, music_extension=".opus")
+            elif musicfile.endswith((".jpg", ".png")):
+                print(f"potential cover art found: {musicfile}.")
             else:
-                print("cover art does not exist. calling stable diffusion api")
-                print(f"sending sd api call:\ntitle: {title_item}\nartist: {artist_item}")
-                if bool(dry_run_enabled) == False:
-                    sd_api_call()
-                elif bool(dry_run_enabled) == True:
-                    api_call_count += 1
-                print(f"Total images created: {api_call_count}")
+                print(f"unsupported file type: {dirpath}/{musicfile}")
                 print("changing directory\n")
-                break
-        elif musicfile.endswith(".jpg" or ".png"):
-            print(f"potential cover art found: {musicfile}.")
-        else:
-            print(f"unsupported file type: {dirpath}/{musicfile}")
-            print("changing directory\n")
 
-print("Scipt complete in %s seconds" % (time.time() - start_time))
-print(f"Total images created: {api_call_count}")
+    print("Scipt complete in %s seconds" % (time.time() - start_time))
+    print(f"Total images created: {api_call_count}")
+
+def argument_parser():
+    parser = argparse.ArgumentParser()
+    parser.add_argument("--config", type=argparse.FileType('r', encoding='UTF-8'))
+    parser.add_argument("--music_directory", type=str, help="Full directory path for your music directory. example: /mnt/data/Music, or C:\\Users\\user\\Music")
+    parser.add_argument("--icon_path", type=str, help="Path to icon. If you use a custom icon, it should be size 150x150. Currently, only .png files supported")
+    parser.add_argument("--sleep_timer", type=int, help="The number of seconds the script will sleep after each image created. Helps with GPU temperatures, but slows the process down")
+    parser.add_argument("--dry_run_enabled", type=bool, help="Skip generating new images, instead will just analyze the music_directory and print results in the console")
+    parser.add_argument("--regenerate_ai_artwork", type=bool, help="Every cover.png in the music_directory or subdirectories that has the 'Author' tag set to 'AI' will be deleted and regenerated")
+    parser.add_argument("--blur_level", type=int, help="0 = No blur effect applied, 5 = high blur. Blur can make SD images appear less ugly at a glance")
+    parser.add_argument("--stable_diffusion_address", type=str, help="Use API calls to public or paid instances at your own risk")
+    
+    parser.add_argument("--prompt", type=str)
+    parser.add_argument("--test_image_output_folder", type=str)

```

Comment:
I believe test_image_output_folder should expect a boolean. In future revisions smeargle could allow for custom directory locations, but currently does not

Reply:
Nevermind I should have just read the next line lol